got-plt/README.md

```diff
@@ -0,0 +1,419 @@
+       0000000000000020  0000000000000008  WA       0     0     8
+[...]
+```
+Both sections are writable and readable (`WA` in the `Flags` column, for *write* and *access*).
+`.got` starts at address `0x200ff8`, `.got.plt` starts at address `0x201000`.
+
+We check their contents:
+```
+$ readelf -x .got main
+
+Hex dump of section `.got`:
+  0x00200ff8 00000000 00000000                   ........
+
+$ readelf -x .got.plt main
+
+Hex dump of section `.got.plt`:
+ NOTE: This section has relocations against it, but these have NOT been applied to this dump.
+  0x00201000 880e2000 00000000 00000000 00000000 .. .............
+  0x00201010 00000000 00000000 76030000 00000000 ........v.......
+```
+
+The `.got` section contains one entry (8 bytes), while the `.got.plt` section contains four entries (32 bytes).
+Both section are zero-filled (with some exceptions that we won't delve into).
+Their useful content (variable and function addresses) is unknown until load time.
+
+We see how the `.got` section is used by the program, in the `.text` section (where the `main()` function resides):
+```
+$ objdump -d -M intel -j .text main
+
+main:     file format elf64-x86-64
+
+
+Disassembly of section .text:
+
+0000000000000380 <_start>:
+ 380:	e8 0c 00 00 00       	call   391 <main>
+ 385:	48 89 c7             	mov    rdi,rax
+ 388:	48 c7 c0 3c 00 00 00 	mov    rax,0x3c
+ 38f:	0f 05                	syscall
+
+0000000000000391 <main>:
+ 391:	55                   	push   rbp
+ 392:	48 89 e5             	mov    rbp,rsp
+ 395:	e8 d6 ff ff ff       	call   370 <flowers@plt>
+ 39a:	e8 d1 ff ff ff       	call   370 <flowers@plt>
+ 39f:	48 8b 05 52 0c 20 00 	mov    rax,QWORD PTR [rip+0x200c52]        # 200ff8 <basket_size>
+ 3a6:	48 c7 00 63 00 00 00 	mov    QWORD PTR [rax],0x63
+ 3ad:	b8 00 00 00 00       	mov    eax,0x0
+ 3b2:	5d                   	pop    rbp
+ 3b3:	c3                   	ret
+```
+
+At address `39f` above, the instruction accesses the `200ff8` address, marked as refering the `basket_size` symbol.
+The `0x200ff8` is the address of the entry in `.got`.
+
+We see how the `.got.plt` section is used by the trampoline code in the `.plt` section:
+```
+$ objdump -d -M intel -j .plt main
+
+main:     file format elf64-x86-64
+
+
+Disassembly of section .plt:
+
+0000000000000360 <.plt>:
+ 360:	ff 35 a2 0c 20 00    	push   QWORD PTR [rip+0x200ca2]        # 201008 <_GLOBAL_OFFSET_TABLE_+0x8>
+ 366:	ff 25 a4 0c 20 00    	jmp    QWORD PTR [rip+0x200ca4]        # 201010 <_GLOBAL_OFFSET_TABLE_+0x10>
+ 36c:	0f 1f 40 00          	nop    DWORD PTR [rax+0x0]
+
+0000000000000370 <flowers@plt>:
+ 370:	ff 25 a2 0c 20 00    	jmp    QWORD PTR [rip+0x200ca2]        # 201018 <flowers>
+ 376:	68 00 00 00 00       	push   0x0
+ 37b:	e9 e0 ff ff ff       	jmp    360 <.plt>
+```
+
+At address `370` above, the instruction accesses the `2010181` address, marked as refering the `flowers` symbol.
+The `0x201018` address is the last (4th) entry in the `.got.plt` section.
```

Comment:
What about "The `0x201018` is the address of the last (4th) entry in the `.got.plt` section."? (I think it is more clear because it avoids confusion between "address of the last entry" and "address stored in the last entry". Also, a similar phrase is used when describing `.got`.)